sql/privileges.go

```diff
@@ -27,6 +27,94 @@ type PrivilegedOperationChecker interface {
 	UserHasPrivileges(ctx *Context, operations ...PrivilegedOperation) bool
 }
 
+// PrivilegeSet is a set containing privileges.
+type PrivilegeSet interface {
```

Comment:
I don't love the idea of exporting these interface definitions. Most of these exported methods are super specific to the implementation of storing / querying these in the privileges package and will never be used outside that context.

The alternative here is to prune these interfaces down to just the set that might be useful outside that package, e.g. `Has` and company. Then this becomes mostly a marker interface, and internally we always do a type conversion to the non-exported type, and we emphasize in the docs that these interfaces are for internal use only and shouldn't be implemented.

Thoughts?